ballerina/json_schema.bal

```diff
@@ -0,0 +1,125 @@
+// Copyright (c) 2025, WSO2 LLC. (https://www.wso2.com).
+//
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+# Defines constraints for JSON schema string type.
+public type JsonStringConstriants record {|
```

Comment:
```suggestion
public type JsonStringConstraints record {|
```

Also, who not just `StringConstraints`? This is already in the jsondata module, so usage will look like `jsondata:StringConstraints`. We try to avoid duplication like `jsondata:JsonStringConstraints`.